R/cal-binary.R

```diff
@@ -0,0 +1,674 @@
+    .data = .data,
+    truth = !!truth,
+    estimate = !!estimate,
+    window_size = window_size,
+    step_size = step_size,
+    conf_level = conf_level,
+    event_level = event_level
+  )
+
+  binary_plot_impl(
+    prob_tbl, predicted_midpoint, event_rate,
+    .data, !!truth, !!estimate,
+    "Predicted Midpoint", "Event Rate",
+    NULL, include_ribbon, include_rug, TRUE
+  )
+}
+
+#' @export
+cal_binary_plot_windowed.data.frame <- cal_binary_plot_windowed_impl
+
+#' @export
+cal_binary_plot_windowed.tune_results <- function(.data,
+                                                  truth = NULL,
+                                                  estimate = NULL,
+                                                  window_size = round(nrow(.data) / 10),
+                                                  step_size = window_size,
+                                                  conf_level = 0.90,
+                                                  include_ribbon = TRUE,
+                                                  include_rug = TRUE,
+                                                  event_level = c("first", "second")) {
+  rs <- tune::collect_predictions(.data, summarize = TRUE)
+
+  te <- tune_results_args(.data, {{ truth }}, {{ estimate }}, event_level, rs)
+
+  cal_binary_plot_windowed_impl(
+    .data = rs,
+    truth = !!te$truth,
+    estimate = !!te$estimate,
+    window_size = window_size,
+    step_size = step_size,
+    conf_level = conf_level,
+    include_ribbon = include_ribbon,
+    include_rug = include_rug,
+    event_level = event_level
+  )
+}
+
+#------------------------------- >> Utils --------------------------------------
+
+binary_plot_impl <- function(tbl, x, y, .data, truth, estimate,
+                             x_label, y_label, sub_title,
+                             include_ribbon, include_rug, include_points) {
+  truth <- enquo(truth)
+  estimate <- enquo(estimate)
+
+
+  x <- enquo(x)
+  y <- enquo(y)
+
+  res <- ggplot(
+    data = tbl,
+    aes(x = !!x)
+  ) +
+    geom_abline(col = "#999999", linetype = 2) +
+    geom_line(aes(y = !!y))
+
+  if (include_points) {
+    res <- res + geom_point(aes(y = !!y))
+  }
+
+
+  if (include_ribbon) {
+    res <- res +
+      geom_ribbon(aes(y = !!y, ymin = lower, ymax = upper), alpha = 0.1)
+  }
+
+  if (include_rug) {
+    level_1_tbl <- dplyr::filter(.data, as.integer(!!truth) == 1)
+    level_2_tbl <- dplyr::filter(.data, as.integer(!!truth) != 1)
+    res <- res +
+      geom_rug(
+        data = level_1_tbl,
+        aes(x = !!estimate, col = !!truth),
+        sides = "b",
+        cex = 0.2,
+        show.legend = FALSE
+      ) +
+      geom_rug(
+        data = level_2_tbl,
+        aes(x = !!estimate, col = !!truth),
+        sides = "t",
+        cex = 0.2,
+        show.legend = FALSE
+      )
+  }
+
+  sub_title <- paste0("'", as_name(truth), "' vs '", as_name(estimate), "'")
+
+  res +
+    tune::coord_obs_pred() +
```

Comment:
Let's use `lims(x = 0:1. y = 0:1)` instead of `coord_obs_pred()`. This will force the whole scale.